### PR TITLE
Add Cucumber Examples as Test Parameters in Allure Report

### PR DESCRIPTION
**Pull request by letsrokk**
[//]: # (
. Thank you so much for sending us a pull request! 
.
. Make sure you have a clear name for your pull request. 
. The name should start with a capital letter and no dot is required in the end of the sentence.
. To link the request with isses use the following notation: (fixes #123, fixes #321\)
.
. An example of good pull request names:
. - Add Russian translation (fixes #123\)
. - Add an ability to disable default plugins
. - Support emoji in test descriptions
)

### Context
[//]: # (
Describe the problem or feature in addition to a link to the issues
)

Right now Cucumber scenarios which are described as Scenario Outlines with Examples are all look the same in report. You can only see parameters when you read Steps descriptions.

before

This pull request uses same approach as TestNG adapter and adds Example values as parameters to Tests in Allure Report

after_2

#### Checklist
- [x] [Sign Allure CLA][cla]

[cla]: https://cla-assistant.io/accept/allure-framework/allure2

### CODE REVIEW

**Comment by eroshenkoam** [comment]
@letsrokk rebase from master, please

**Comment by baev** [comment]
See http://blog.steveklabnik.com/posts/2012-11-08-how-to-squash-commits-in-a-github-pull-request and http://stackoverflow.com/questions/14534397/how-to-squash-all-my-commits-into-one-github

**Comment by letsrokk** [comment]
ok, sure, will do

File: allure-cucumber-jvm/src/main/java/io/qameta/allure/cucumberjvm/AllureCucumberJvm.java [hunk 1]
```diff
@@ -106,6 +114,24 @@ public void startOfScenarioLifeCycle(final Scenario scenario) {
 
     }
 
+    private List<Parameter> getExampleAsParameters(final Scenario scenario, final Examples exmpls) {
+        if (exmpls != null) {
```
Comment (baev) [inline allure-cucumber-jvm/src/main/java/io/qameta/allure/cucumberjvm/AllureCucumberJvm.java]: if (Objects.isNull(exmpls)) {
    return ...
}
...

File: allure-cucumber-jvm/src/main/java/io/qameta/allure/cucumberjvm/AllureCucumberJvm.java [hunk 2]
```diff
@@ -106,6 +114,24 @@ public void startOfScenarioLifeCycle(final Scenario scenario) {
...
+            final List<String> names = exmpls.getRows().get(0).getCells();
```
Comment (baev) [inline allure-cucumber-jvm/src/main/java/io/qameta/allure/cucumberjvm/AllureCucumberJvm.java]: can example be empty?

File: allure-cucumber-jvm/src/main/java/io/qameta/allure/cucumberjvm/AllureCucumberJvm.java [hunk 3]
```diff
@@ -106,6 +114,24 @@ public void startOfScenarioLifeCycle(final Scenario scenario) {
...
+
+            final List<String> values = exmpls.getRows().stream()
+                    .filter(exmpl -> Objects.equals(exmpl.getLine(), scenario.getLine()))
+                    .findFirst()
+                    .get()
```
Comment (baev) [inline allure-cucumber-jvm/src/main/java/io/qameta/allure/cucumberjvm/AllureCucumberJvm.java]: possible NPE?

**Comment by letsrokk** [comment]
Closing as outdated. Cucumber JVM released new 2.x version with incompatible Formatters

**Pull request closed without merging**